modules/core/src/test/scala/com.snowplowanalytics.iglu.client/resolver/ResolverResultSpec.scala

```diff
@@ -0,0 +1,521 @@
+    } yield (response1, response2, response3)
+
+    val (state, (response1, response2, response3)) =
+      result.run(ResolverSpecHelpers.RegistryState.init).value
+
+    val firstFailed = response1 must beLeft[ResolutionError].like { case ResolutionError(history) =>
+      history must haveValue(
+        LookupHistory(Set(RegistryError.RepoFailure("shouldn't matter")), 1, time)
+      )
+    }
+    val secondFailed = response2 must beLeft[ResolutionError].like {
+      case ResolutionError(history) =>
+        history must haveValue(
+          LookupHistory(Set(RegistryError.RepoFailure("shouldn't matter")), 1, time)
+        )
+    }
+
+    val thirdSucceeded = response3 must beRight[SchemaLookupResult].like {
+      case ResolverResult.Cached(key, value, _) =>
+        key must beEqualTo(schemaKey) and (value must beEqualTo(Json.Null))
+    }
+    val requestsNumber = state.req must beEqualTo(2)
+
+    firstFailed and secondFailed and thirdSucceeded and requestsNumber
+  }
+
+  def e7 = {
+    val schemaKey =
+      SchemaKey(
+        "com.snowplowanalytics.iglu-test",
+        "future_schema",
+        "jsonschema",
+        SchemaVer.Full(1, 0, 0)
+      )
+    val correctSchema = Json.Null
+    val responses = List(
+      RegistryError.RepoFailure("Timeout exception 1").asLeft,
+      RegistryError.RepoFailure("Timeout exception 2").asLeft,
+      correctSchema.asRight,
+      RegistryError.RepoFailure("Should never be reached").asLeft
+    )
+
+    val httpRep =
+      Registry.Http(Registry.Config("Mock Repo", 1, List("com.snowplowanalytics.iglu-test")), null)
+
+    implicit val cache          = ResolverSpecHelpers.staticCache
+    implicit val cacheList      = ResolverSpecHelpers.staticCacheForList
+    implicit val clock          = ResolverSpecHelpers.staticClock
+    implicit val registryLookup = ResolverSpecHelpers.getLookup(responses, Nil)
+
+    val result = for {
+      resolver <-
+        Resolver
+          .init[StaticLookup](
+            10,
+            Some(1000.seconds),
+            httpRep
+          )
+      _      <- resolver.lookupSchemaResult(schemaKey)
+      _      <- StaticLookup.addTime(2.seconds)
+      _      <- resolver.lookupSchemaResult(schemaKey)
+      _      <- StaticLookup.addTime(2.seconds)
+      _      <- resolver.lookupSchemaResult(schemaKey)
+      _      <- StaticLookup.addTime(2.seconds)
+      result <- resolver.lookupSchemaResult(schemaKey) // ... but don't try to overwrite it
+    } yield result
+
+    val (state, response) = result.run(ResolverSpecHelpers.RegistryState.init).value
+
+    // Final response must not overwrite a successful one
+    val finalResult = response must beRight[SchemaLookupResult].like {
+      case ResolverResult.Cached(key, value, _) =>
+        key must beEqualTo(schemaKey) and (value must beEqualTo(Json.Null))
+    }
+
+    // Check that it attempted to get fourth schema (500 response)
+    val lookupTries = state.req must beEqualTo(4)
```

Comment:
I think I'm missing something here - I can't work out why we would expect there to be a fourth request if the third request was successful and isn't stale.

Reply:
[This](https://github.com/snowplow/iglu-scala-client/pull/208/files#diff-15822dfa15d03bb4368de73705f1e90a532f64de8b958069433643649caa1a4aR242) is the same test in the 1.2.0 release

Reply:
Ooops, I was convinced that the TTL was 1000 seconds but Piotr nudged me to look at the [analogous test](https://github.com/snowplow/iglu-scala-client/blob/master/modules/core/src/test/scala/com.snowplowanalytics.iglu.client/resolver/ResolverSpec.scala#L243) in `ResolverSpec` and this makes a lot more sense now.